File: frontend/src/types/submission.ts
export interface Submission {
  id: number
  user_id: number
  submission_type: 'meme' | 'music'
  status: 'pending' | 'approved' | 'rejected'
  title: string
  description: string | null
  file_url: string
  source_type: 'upload' | 'video_upload' | 'douyin'
  content_id: number | null
  content_deleted: boolean
  content_is_featured?: boolean
  cover_url: string | null
  character_id: number | null
  character_ids: number[]
  source_name: string | null
  source_url: string | null
  source_author: string | null
  author_name: string | null
  author_uid: string | null
  reject_reason: string | null
  reviewed_by: number | null
  reviewed_at: string | null
  created_at: string
  updated_at: string
}

File: backend/app/schemas/submission.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class SubmissionRead(BaseModel):
    id: int
    user_id: int
    submission_type: str
    status: str
    title: str
    description: str | None = None
    file_url: str
    source_type: str
    content_id: int | None = None
    content_deleted: bool
    cover_url: str | None = None
    character_id: int | None = None
    character_ids: list[int] = []
    source_name: str | None = None
    source_url: str | None = None
    source_author: str | None = None
    author_name: str | None = None
    author_uid: str | None = None
    reject_reason: str | None = None
    reviewed_by: int | None = None
    reviewed_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SubmissionUpdate(BaseModel):
    title: str = Field(min_length=1, max_length=160)
    description: str | None = None
    character_ids: list[int] = Field(default_factory=list)
    source_name: str | None = Field(default=None, max_length=120)
    source_url: str | None = Field(default=None, max_length=500)
    source_author: str | None = Field(default=None, max_length=100)
